cmd/destroy/local.go

```diff
@@ -0,0 +1,415 @@
+package destroy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+	"time"
+
+	pipelineCMD "github.com/okteto/okteto/cmd/pipeline"
+	"github.com/okteto/okteto/cmd/utils"
+	"github.com/okteto/okteto/cmd/utils/executor"
+	"github.com/okteto/okteto/pkg/analytics"
+	"github.com/okteto/okteto/pkg/cmd/pipeline"
+	"github.com/okteto/okteto/pkg/constants"
+	oktetoErrors "github.com/okteto/okteto/pkg/errors"
+	"github.com/okteto/okteto/pkg/format"
+	"github.com/okteto/okteto/pkg/k8s/namespaces"
+	"github.com/okteto/okteto/pkg/k8s/secrets"
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/model"
+	"github.com/okteto/okteto/pkg/okteto"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/client-go/kubernetes"
+)
+
+type localDestroyCommand struct {
+	*localDestroyAllCommand
+	manifest *model.Manifest
+}
+
+type localDestroyAllCommand struct {
+	configMapHandler  configMapHandler
+	nsDestroyer       destroyer
+	executor          executor.ManifestExecutor
+	oktetoClient      *okteto.OktetoClient
+	secrets           secretHandler
+	k8sClientProvider okteto.K8sClientProvider
+}
+
+func newLocalDestroyer(
+	manifest *model.Manifest,
+	destroyerAll *localDestroyAllCommand,
+) *localDestroyCommand {
+	return &localDestroyCommand{
+		destroyerAll,
+		manifest,
+	}
+}
+
+func newLocalDestroyerAll(
+	k8sClient *kubernetes.Clientset,
+	k8sClientProvider okteto.K8sClientProvider,
+	executor executor.ManifestExecutor,
+	nsDestroyer destroyer,
+	oktetoClient *okteto.OktetoClient,
+) *localDestroyAllCommand {
+	return &localDestroyAllCommand{
+		configMapHandler:  newConfigmapHandler(k8sClient),
+		secrets:           secrets.NewSecrets(k8sClient),
+		k8sClientProvider: k8sClientProvider,
+		oktetoClient:      oktetoClient,
+		nsDestroyer:       nsDestroyer,
+		executor:          executor,
+	}
+}
+
+func (ld *localDestroyCommand) destroy(ctx context.Context, opts *Options) error {
+
+	if opts.RunInRemote {
+		if ld.manifest.Deploy != nil {
+			if ld.manifest.Deploy.Image == "" {
+				ld.manifest.Deploy.Image = constants.OktetoPipelineRunnerImage
+			}
+		}
+	}
+
+	err := ld.runDestroy(ctx, opts)
+	if err == nil {
+		oktetoLog.Success("Development environment '%s' successfully destroyed", opts.Name)
+	}
+	analytics.TrackDestroy(err == nil, opts.DestroyAll)
+
+	return err
+}
+
+func (dc *localDestroyAllCommand) destroy(ctx context.Context, opts *Options) error {
```

Comment:
NIT: Should we move this to another file?

Reply:
I will!